### ISSUE TITLE
Resolving bracket links to explicit heading identifiers?

### ISSUE THREAD

**Issue by iandol** [jgm/pandoc#6085]
Say that I add an explicit heading ID `{#conclusion}` to a heading, I currently cannot use an implicit link  like `[conclusion][]` or `[Summary][conclusion]' to that heading, it fails to be parsed by Pandoc as a valid link:

```html
➜ pandoc
See [conclusion][], [Summary][conclusion] or [Summary](#conclusion) for more.

# §3 Discussion {#conclusion}
Lørem ipsum dolør sit amet.

<p>See [conclusion][], [Summary][conclusion] or <a href="#conclusion">Summary</a> for more.</p>
<h1 id="conclusion">§3 Discussion</h1>
<p>Lørem ipsum dolør sit amet.</p>
```

Sample document:

```markdown
---
title: Cross Referencing Test
author: Joanna Doe
---

# §1 Introduction
 
Please see [conclusion][] for more. Please see [discussion](#conclusion) for more.

# §2 Results

## §2.1 Sun

Lørem ipsum dolør sit amet.  

## §2.2 Moon

Lørem ipsum dolør sit amet.  

# §3 Discussion {#conclusion}

Lørem ipsum dolør sit amet.  
```

I can use an internal link style `[text](#id)` and that works, but there are some workflow cases where an implicit style `[link]` would be better. What is the reason we can't link to the explicit #ID in cases like these?

**Comment by iandol** [comment]
Note that cross-references to explicit header IDs works fine with MultiMarkDown:

```markdown
➜ multimarkdown
Please see [Summary][conclusion] or [Summary](#conclusion) for more.

# §3 Discussion [conclusion]

Lørem ipsum dolør sit amet.
```
```html
<p>Please see <a href="#conclusion">Summary</a> or <a href="#conclusion">Summary</a> for more.</p>

<h1 id="conclusion">§3 Discussion</h1>

<p>Lørem ipsum dolør sit amet.</p>
```

**Comment by mb21** [comment]
You would need to use the same text as in the heading, not the id:

```
See [conclusion][§3 Discussion], [Summary][§3 Discussion] or [Summary](#conclusion) for more.

# §3 Discussion {#conclusion}
Lørem ipsum dolør sit amet.

```

**Comment by iandol** [comment]
@mb21 – for this particular workflow we **_cannot_** use the heading text, that is why I am asking to allow the use of explicit IDs.

The reason we cannot use text? In the authoring environment [Scrivener](http://literatureandlatte.com/), its compiler can add prefixes and suffixes to headings, and these can be generated at compile time (after the author has finished writing). That means the author cannot know what the exact heading text will be. Scrivener allows you to change the text, so for example:

`Discussion` becomes  `# §IV – Discussion #` during compilation, and Scrivener can rewrite links to support the use of prefix text only, for example `please refer to chapter [§IV][§IV – Discussion]` — however Scrivener cannot write the correct link ID in these cases (as Pandoc will [parse and change the heading text to create auto-IDs](https://pandoc.org/MANUAL.html#extension-auto_identifiers), thus breaking the Scrivener compiled links) . If the user can add an explicit ID that is unique and unchanging, then this problem can be solved.

In summary this authoring workflow breaks cross-referencing for Scrivener+Pandoc users as Pandoc does not currently support ID linking. MMD users have no problem (as mentioned it can use explicit IDs as link IDs), supporting the idea that if the user explicitly adds an ID, allowing it use as a reference in a link is not out of scope.

**Comment by mb21** [comment]
But what should the following result in:

```
# foo {#bar}

# bar

what does this [link point to][bar]?
```

It doesn't seem to make sense that sometimes you should put in the link text and sometimes the id... and all to accommodate the edge case/combination of Scrivener and pandoc. Especially since there is a workaround with `[link](#explicit-id)`...

But feel free to open a discussion on the [pandoc-discuss mailing list](https://groups.google.com/d/forum/pandoc-discuss)...

**Comment by iandol** [comment]
@mb21 ok thanks for your inputs, and for your general work on improving Pandoc! 

The `[text](#id)` links cannot work in this case as they are not "cleaned" by Pandoc like the `[text][id]` links are. 

My assumptions for precedence is that explicit links should be resolved if there is a conflict. I'll open a discussion about this over on the group...
